client/src/app.jsx

```diff
@@ -1,19 +1,167 @@
+      brands: itemBrands
+    });
+  }
+
+  //get price
+  getPrices () {
+    let multipleIds = { ids: this.state.ids};
+    axios.post('http://localhost:4003/priceandinventory/id/multiple', multipleIds.ids)
+      .then(response => {
+        let itemPrices = [];
+        for (let i = 0; i < response.data.length; i++) {
+          itemPrices.push(response.data[i].price);
+        }
+        this.setState({prices: itemPrices});
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  }
+
+  //get photos
+  getPhotos () {
+    let productIds = { ids: this.state.ids};
+    axios.post('http://localhost:4002/photos/product/primary/multiple', productIds.ids)
+      .then(response => {
+        let itemPhotos = Object.values(response.data);
+        this.setState({
+          photos: itemPhotos
+        });
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  }
+
+  //get ratings
+  getRatings () {
+    let productIds = this.state.ids;
+    for (let i = 0; i < productIds.length; i++) {
+      axios.get(`http://localhost:4006/Reviews/getReviewSummary/${productIds[i]}`)
+        .then(response => {
+          let numOfRatings = response.data.totalRatings;
+          let starRating = response.data.averageRating;
+          starRating = parseInt(starRating);
+          let newRatings = this.state.ratings;
+          newRatings.push(numOfRatings);
+          let newStarRatings = this.state.starRatings;
+          newStarRatings.push(starRating);
+          this.setState({
+            ratings: newRatings,
+            starRatings: newStarRatings
+          });
+        })
+        .catch(err => {
+          console.log(err);
+        });
+    }
+  }
+
+
+  //get items from comparison/getSimilarItems/comparison
+  getSimilarItems (id) {
+    let productId = id;
+    let productIds = [];
+    productIds.push(productId);
+    this.setState({
+      ids: productIds
+    });
+    axios.get(`http://localhost:4005/comparison/getSimilarItems/${productId}`)
+      .then((response) => {
+        let similarIds = response.data;
+        this.setState({
+          ids: productIds.concat(similarIds)
+        });
+        this.getItemDescription(this.state.ids);
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  }
+
+  //get info for each item by sending request to description service, price service
+
+
+  //componentDidMount shall call getSimilarItems
+  componentDidMount () {
+    //get productId from url or 1000
+    let url = window.location.href;
+    let productId = url.split('/')[3] || 1000;
+    //call getSimilarItems with productId
+    this.getSimilarItems(productId);
   }
 
   render () {
     return (
       <div>
         <Title>Comparison Service</Title>
+        <ItemList names={this.state.names} brands={this.state.brands} prices={this.state.prices} photos={this.state.photos} ratings={this.state.ratings} starRatings={this.state.starRatings}></ItemList>
       </div>
     );
   }
```

Comment:
Despite making several axios requests to multiple services, you code is well formatted and easy to follow! My feedback would be to make sure you're using the right HTTP request method.

Reply:
Got it!